ddtrace/contrib/urllib3/patch.py

```diff
@@ -0,0 +1,153 @@
+import urllib3
+
+from ddtrace import config
+from ddtrace.pin import Pin
+from ddtrace.vendor.wrapt import wrap_function_wrapper as _w
+
+from .. import trace_utils
+from ...compat import parse
+from ...constants import ANALYTICS_SAMPLE_RATE_KEY
+from ...ext import SpanTypes
+from ...ext import http
+from ...propagation.http import HTTPPropagator
+from ...utils.formats import asbool
+from ...utils.formats import get_env
+from ...utils.http import sanitize_url_for_tag
+from ...utils.wrappers import unwrap as _u
+
+
+# Ports which, if set, will not be used in hostnames/service names
+DROP_PORTS = (80, 443)
+
+# Initialize the default config vars
+config._add(
+    "urllib3",
+    {
+        "_default_service": "urllib3",
+        "distributed_tracing": asbool(get_env("urllib3", "distributed_tracing", default=True)),
+        "analytics_enabled": asbool(get_env("urllib3", "analytics_enabled", default=False)),
+        "analytics_sample_rate": float(get_env("urllib3", "analytics_sample_rate", default=1.0)),
+        "trace_query_string": asbool(get_env("urllib3", "trace_query_string", default=False)),
+        "split_by_domain": asbool(get_env("urllib3", "split_by_domain", default=False)),
+    },
+)
+
+
+def patch():
+    """Enable tracing for all urllib3 requests"""
+    if getattr(urllib3, "__datadog_patch", False):
+        return
+    setattr(urllib3, "__datadog_patch", True)
+
+    _w("urllib3", "connectionpool.HTTPConnectionPool.urlopen", _wrap_urlopen)
+
+
+def unpatch():
+    """Disable trace for all urllib3 requests"""
+    if getattr(urllib3, "__datadog_patch", False):
+        setattr(urllib3, "__datadog_patch", False)
+
+        _u(urllib3.connectionpool.HTTPConnectionPool, "urlopen")
+
+
+def _infer_argument_value(args, kwargs, pos, kw, default=None):
```

Comment:
This would be a good helper to have in trace_utils to be used across integrations. Might be worth doing in a separate PR but I'm ok to leave it here too. It should probably have a few unit tests as well.